Packages/MIES/MIES_SweepFormula.ipf

```diff
@@ -411,37 +426,100 @@ Function/WAVE FormulaExecutor(jsonID, [jsonPath, graph])
 			MatrixOP/FREE out = (sumCols(magSqr(wv - rowRepeat(averageCols(wv), numRows(wv))))/(numRows(wv) - 1))^t
+			FormulaWaveScaleTransfer(wv, out, COLS, ROWS)
 			break
 		case "stdev":
 			ASSERT(DimSize(wv, LAYERS) <= 1, "Unhandled dimension")
 			ASSERT(DimSize(wv, CHUNKS) <= 1, "Unhandled dimension")
 			MatrixOP/FREE out = (sqrt(sumCols(powR(wv - rowRepeat(averageCols(wv), numRows(wv)), 2))/(numRows(wv) - 1)))^t
+			FormulaWaveScaleTransfer(wv, out, COLS, ROWS)
 			break
 		case "derivative":
 			Make/FREE out
 			if(DimSize(wv, COLS) == 0)
 				ASSERT(DimSize(wv, ROWS) > 1, "Can not differentiate single point waves")
 				Differentiate/DIM=(ROWS) wv/D=out
+				CopyScales wv, out
+				SetScale/P x, DimOffset(wv, ROWS), DimDelta(wv, ROWS), "d/dx", out
 			else
 				ASSERT(DimSize(wv, COLS) > 1, "Can not differentiate single point waves")
 				Differentiate/DIM=(COLS) wv/D=out
+				CopyScales wv, out
+				SetScale/P y, DimOffset(wv, COLS), DimDelta(wv, COLS), "d/dx", out
 			endif
 			break
 		case "integrate":
 			Make/FREE out
 			if(DimSize(wv, COLS) == 0)
 				ASSERT(DimSize(wv, ROWS) > 1, "Can not integrate single point waves")
 				Integrate/METH=1/DIM=(ROWS) wv/D=out
+				CopyScales wv, out
+				SetScale/P x, DimOffset(wv, ROWS), DimDelta(wv, ROWS), "dx", out
 			else
 				ASSERT(DimSize(wv, COLS) > 1, "Can not integrate single point waves")
 				Integrate/METH=1/DIM=(COLS) wv/D=out
+				CopyScales wv, out
+				SetScale/P y, DimOffset(wv, COLS), DimDelta(wv, COLS), "dx", out
 			endif
 			break
+		case "time":
+		case "xvalues":
+			Make/FREE/N=(DimSize(wv, ROWS), DimSize(wv, COLS), DimSize(wv, LAYERS), DimSize(wv, CHUNKS)) out = DimOffset(wv, ROWS) + p * DimDelta(wv, ROWS)
+			break
+		case "setscale":
+			/// `setscale(data, dim, dimOffset, dimDelta[, unit])
+			numIndices = JSON_GetArraySize(jsonID, jsonPath)
+			ASSERT(numIndices < 6, "Maximum number of arguments exceeded.")
+			ASSERT(numIndices > 1, "At least two arguments.")
+			WAVE data = FormulaExecutor(jsonID, jsonPath = jsonPath + "/0", graph = graph)
+			WAVE/T dimension = FormulaExecutor(jsonID, jsonPath = jsonPath + "/1")
+			ASSERT(DimSize(dimension, ROWS) == 1 && GrepString(dimension[0], "[x,y,z,t]") , "undefined input for dimension")
+
+			if(numIndices >= 3)
+				WAVE offset = FormulaExecutor(jsonID, jsonPath = jsonPath + "/2")
+				ASSERT(DimSize(offset, ROWS) == 1, "wrong usage of argument")
+			else
+				Make/FREE/N=1 offset  = {0}
+			endif
+			if(numIndices >= 4)
+				WAVE delta = FormulaExecutor(jsonID, jsonPath = jsonPath + "/3")
+				ASSERT(DimSize(delta, ROWS) == 1, "wrong usage of argument")
+			else
+				Make/FREE/N=1 delta = {1}
+			endif
+			if(numIndices == 5)
+				WAVE/T unit = FormulaExecutor(jsonID, jsonPath = jsonPath + "/4")
+				ASSERT(DimSize(unit, ROWS) == 1, "wrong usage of argument")
+			else
+				Make/FREE/N=1/T unit = {""}
+			endif
+
+			strswitch(dimension[0])
+				case "x":
+					SetScale x, offset[0], delta[0], unit[0], data
```

Comment:
weired bug. Probably wrong data usages.

Use setscale(range(0…10), 0, 1)

@t-b @MichaelHuth 

If `delta[0] == 1` or any other unsigned integer, I can use `print delta[0]` to output 1.
So print outputs the right thing but the wave is scaled to negative values like `-1.6` for 1 and `-1.7` for 5 if I remember correctly. That could be some wrongly interpreted data format. I stored the wave data in a variable and it behaves similar. Also using str2num(num2str()) to clean up gives same result.

Reply:
> If you do not use the /P flag, SetScale converts num1  and num2  into a starting index value and a delta index value

Are you sure this should scale on the index values?

Reply:
I have added the `/P` flag and now it works for `ROWS` but `COLS` is still broken.

I did not quite get what the difference between omitting `/P` and explicitly giving `/P` is.

Reply:
Tracked in #295.